tutorials/functional-full-stack/purescript-bridge/tutorial.md

```diff
@@ -0,0 +1,384 @@
+At [Stack Builders](https://www.stackbuilders.com) we are working on a full-stack app with [CollegeVine](https://www.collegevine.com/) using Functional Languages.
+We have a [Haskell](https://www.haskell.org) backend written in [Servant](http://haskell-servant.readthedocs.io) that manipulates the database and offers some endpoints to a [PureScript](http://www.purescript.org) frontend,
+that does all the React-like magic to show a really nice interface on the user's browser.
+It's great because we have advanced types, purity and all the awesome benefits that offers the Functional World.
+But not everything is perfect.
+
+### Motivation
+The problem is that we have two different codebases: one in Haskell and the other in PureScript.
+The syntax is almost the same, but not exactly the same. For example, on Haskell we do:
+```Haskell
+{-# LANGUAGE DeriveGeneric #-}
+
+data Blah = Blah
+  {
+    bFoos :: [Foo]
+  } deriving (Generic)
+```
+Whereas if I want to get a Generic instance for my `Blah` type, in PureScript I have to do:
+
+```Haskell
+data Blah = Blah
+  {
+    bFoos :: Array Foo
+  }
+
+derive instance genericBlah :: Generic Blah
+```
+It's almost the same, but I bet you a penny that Haskell will not accept that `derive instance` declaration.
+Also, in PureScript we import typeclasses using `import MyModule (class MyClass)`,
+and we initialize records with `Constructor {field : value}`, whereas in Haskell it's `import MyModule (MyClass)` and `Constructor {field = vale}`.
+Tiny, but huge differences.
+
+At the end of the day, that means we can't directly share files.
+So we have types that describe the entities in the backend,
+and we have the same types that describe the same entities in the frontend, in different files.
+And every now and then, the backend team changes something on the backend types,
+but forgets to change it on the front end types.
+Suddenly we start to get the dreaded runtime errors,
+because the frontend is no longer able to decode the data sent from the backend,
+because that data no longer conforms to the standard the frontend expects.
+
+Ideally, we would separate the common types to some files, and use these files in the backend and the frontend.
+Then, the frontend inevitably follows the changes in types from the backend,
+and refuses to compile if the change is too big and a developer has to look at it.
+But, again, Haskell code is almost like PureScript code, but not completely compatible.
+So this is not possible.
+
+### The next best thing
+Well, if we can't use the same files, we have to look for something not that far from that ideal world.
+If we can somehow automatically generate PureScript data types from Haskell data types,
+we could prevent the problem of type difference. We would effectively extend the typesystem from the backend to the frontend.
+
+### `purescript-bridge` to the rescue
+Turns out this idea is not new, and Robert Klotzner has already done it for us, which is quite nice.
```

Comment:
Can yo reference Robert Klotzner with a link to his personal website of Twitter account?